src/Elmish.WPF.Tests/ViewModelTests.fs

```diff
@@ -1062,11 +1063,11 @@ module SubModel =
       let binding = subModel name getModel toMsg [] sticky
       let vm = TestVm(m1, binding)
 
-      test <@ (vm.Get name : ViewModel<obj, obj>).CurrentModel |> unbox = (getModel m1).Value @>
+      test <@ (vm.Get name : ViewModel<int, obj>).CurrentModel |> unbox = (getModel m1).Value @>
```

Comment:
Why this change?

Reply:
This was necessary because of the casting issues. We are now retaining more type information in sub view models (because we need the actual type around, and can't just wrap it in a `ViewModel<obj, obj>` anymore)

Reply:
See my [comments here in context](https://github.com/elmish/Elmish.WPF/pull/449/files#r834897266)

Reply:
Ok.

There (and similar places in this file), the `|> unbox` can be removed now.

Reply:
Good point, I didn't think about that

Reply:
In about an hour, I will make that change (if you haven't already) and then merge this PR.

Reply:
I pushed already. Make sure I caught all of them.

Reply:
Oh, I saw you removed many and merged without double checking.  It is ok if we missed some.  Getting most of them in this commit helps to show that these changes are related.  We can remove any additional ones later.